lib/common/metric/MetricFactory.ts

```diff
@@ -54,6 +54,11 @@ export interface MetricFactoryProps {
   readonly globalDefaults?: MetricFactoryDefaults;
 }
 
+export enum ResolvePeriodHandling {
+  ERROR,
+  CLAMP_TO_MINIMUM_REPORT_FREQUENCY,
```

Comment:
Can you add some docs to describe the behaviours a bit more?

That said, it seems safer to always error and force the implementer to be explicit with their intent on what the period should be.